### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by TRoZZNL**
Changes in this commit:

- Some additional Async Methods introduced.
- Added API Key setting to environment.
- Added Authentication API.
- Added Navigation API
- Added Swagger docs to API

### CODE REVIEW

**Review by bart-van-der-drift — APPROVED** [review]

File: src/Sushi.Mediakiwi.API/Controllers/Authentication.cs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+namespace Sushi.Mediakiwi.API.Controllers
+{
+    [ApiController]
+    [MediakiwiApiAuthorize]
+    [Route(Common.MK_CONTROLLERS_PREFIX + "authentication")]
+    [ResponseCache(NoStore = true, Location = ResponseCacheLocation.None)]
+    public class Authentication : ControllerBase
+    {
+        private readonly IUserService userService;
```
Comment (mark-devries) [inline src/Sushi.Mediakiwi.API/Controllers/Authentication.cs:18]: rename to _userService, convention for private fields

File: src/Sushi.Mediakiwi.API/Controllers/Authentication.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                return Unauthorized();
+            }
+
+            // Check for user existence and create jwtToken if valid.
+            LoginResponse result = await userService.LoginAsync(request).ConfigureAwait(false);
+
+            // When successful, return filled response object
+            if (result.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                var cookieOptions = new CookieOptions
```
Comment (mark-devries) [inline src/Sushi.Mediakiwi.API/Controllers/Authentication.cs:61]: use cookie policy middleware to set cookie options for all cookies:
https://docs.microsoft.com/en-us/aspnet/core/security/authentication/cookie?view=aspnetcore-6.0#cookie-policy-middleware

File: src/Sushi.Mediakiwi.API/Controllers/Navigation.cs [hunk 3]
```diff
@@ -0,0 +1,820 @@
...
+     //       #region Tabs
+
+            if (Resolver.ListInstance?.wim?.Page?.HideTabs != true)
+            {
+                return null;
+            }
+
+            string tabTag = null;
+            
+   //         bool isPageProperty = resolver.List.Type == Data.ComponentListType.PageProperties;
```
Comment (mark-devries) [inline src/Sushi.Mediakiwi.API/Controllers/Navigation.cs:211]: remove commented code

File: src/Sushi.Mediakiwi.API/Extensions/Extensions.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                QueryString = existingRequestFeature.QueryString,
+                RawTarget = existingRequestFeature.RawTarget,
+                Headers = new HeaderDictionary(requestHeaders),
+            };
+
+            if (copyBody)
+            {
+                // We need to buffer first, otherwise the body won't be copied
+                // Won't work if the body stream was accessed already without calling EnableBuffering() first or without leaveOpen
+                httpContext.Request.EnableBuffering();
```
Comment (mark-devries) [inline src/Sushi.Mediakiwi.API/Extensions/Extensions.cs:48]: this is a performance killer, don't do this

why do we need to clone the httpcontext?

File: src/Sushi.Mediakiwi.API/Controllers/Authentication.cs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+            // Compare API key with environment
+            var env = await Data.Environment.SelectOneAsync().ConfigureAwait(false);
+            if (env.ApiKey.Equals(request.ApiKey, StringComparison.InvariantCulture) == false)
+            {
+                return Unauthorized();
+            }
+
+            // Check for user existence and create jwtToken if valid.
+            LoginResponse result = await userService.LoginAsync(request).ConfigureAwait(false);
```
Comment (mark-devries) [inline src/Sushi.Mediakiwi.API/Controllers/Authentication.cs:56]: this is the preferred method of using cookie authentication:
https://docs.microsoft.com/en-us/aspnet/core/security/authentication/cookie?view=aspnetcore-6.0#create-an-authentication-cookie

File: src/Sushi.Mediakiwi.API/Controllers/Authentication.cs (hunk 1 above)
Comment (TRoZZNL) [reply src/Sushi.Mediakiwi.API/Controllers/Authentication.cs:18]: Agreed, adjusted

**Pull request merged**